lib/recommendation-server.js

```diff
@@ -71,14 +106,21 @@ RecommendationServer.prototype = {
     }
   },
   search: function(payload) {
+    let server = this;
     if (!payload.query) {
       return;
     }
-    this._search(payload.query).then(data => {
-      this.events.publish('recommendation', data);
+    server._search(payload.query).then(data => {
+      if (server._resultHasChanged(data)) {
```

Comment:
Hmm. Now that I'm staring at this, it might be easier to just inline the object equality check?

`if (this.objectUtils.deepEqual(this.prev, data)) { ...`